java/AvroOneTopicManySubjects/flink-consumer/src/main/java/com/amazonaws/services/msf/Option.java

```diff
@@ -0,0 +1,26 @@
+package com.amazonaws.services.msf;
+
+/**
+ * Option type is a generic container to hold deserialized records.
+ * It can be useful to introduce fields into this type to handle partitioning
+ * strategies and event time extraction. However, for those scenarios to work
+ * all subjects should have a standard set of fields.
+ */
+class Option {
```

Comment:
I understand to goal of this class now. I got confused because "option" is not really a good name.
An "option" is normally a container which can be either "something" or "nothing". Like  Java `Optional` or Scala `Option`
This is generic single-element container which contains whatever. More a "box" or "container" something similar

Reply:
Have you checked this does not fallback to Kryo for serialization? It could become a performance killer.
I'd suggest a unit test like [this one](https://github.com/aws-samples/amazon-managed-service-for-apache-flink-examples/blob/main/java/Serialization/CustomTypeInfo/src/test/java/com/amazonaws/services/msf/domain/VehicleEventSerializationTest.java)